### PR TITLE
Code Modification in Yash Datalake

### PR DESCRIPTION
**Pull request by jashneet-bhatia**
# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: yash-datalake/azuredeploy.json [hunk 1]
```diff
@@ -213,7 +219,7 @@
     {
       "type": "Microsoft.Resources/deployments",
       "name": "storageAccount",
-      "apiVersion": "2019-09-01",
+      "apiVersion": "2019-08-01",
```
Comment (bmoore-msft) [inline yash-datalake/azuredeploy.json]: why are we rolling back to older apiVersions?

File: yash-datalake/nested/blobTriggerdFunction.json [hunk 2]
```diff
@@ -112,7 +112,7 @@
     {
       "type": "Microsoft.Web/serverfarms",
       "name": "[parameters('serverfarms_name')]",
-      "apiVersion": "2019-08-01",
+      "apiVersion": "2018-11-01",
```
Comment (bmoore-msft) [inline yash-datalake/nested/blobTriggerdFunction.json]: same as above, why are we rolling back the apiVersion

File: yash-datalake/nested/dataFactory.json [hunk 3]
```diff
@@ -84,13 +84,12 @@
             "[parameters('dataFactoryName')]"
           ],
           "properties": {
-            "type": "AzureStorage",
-            "description": "Azure Storage linked service",
+            "annotations": [],
```
Comment (bmoore-msft) [inline yash-datalake/nested/dataFactory.json]: remove empty properties - see ci results for the complete list

File: yash-datalake/nested/storageAccount.json [hunk 4]
```diff
@@ -31,7 +31,7 @@
     {
       "type": "Microsoft.Storage/storageAccounts",
       "name": "[parameters('storageAccountName')]",
-      "apiVersion": "2019-06-01",
+      "apiVersion": "2018-11-01",
```
Comment (bmoore-msft) [inline yash-datalake/nested/storageAccount.json]: why the rollback?

File: yash-datalake/azuredeploy.json [hunk 5]
```diff
@@ -147,6 +147,13 @@
       },
       "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datafactorya'))]"
     },
+    "appName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the Azure data lake UI app registered. Must be globally unique."
+      },
+      "defaultValue": "[toLower(uniqueString(resourceGroup().id, 'azuredatalakeuiappa'))]"
```
Comment (bmoore-msft) [inline yash-datalake/azuredeploy.json]: the CI failure for this run was due to the eventhub name starting with a number... we should flip the uniqueString functions to start with the literal string and the unique string appended... e.g.

  "defaultValue": "[toLower(concat('azuredatalakeuiappa, uniqueString(resourceGroup().id)))]", 

You'll have to check for max length on the names for this as well...

File: yash-datalake/nested/dataFactory.json (hunk 3 above)
Comment (jashneet-bhatia) [reply yash-datalake/nested/dataFactory.json]: Changed as requested

File: yash-datalake/nested/blobTriggerdFunction.json (hunk 2 above)
Comment (jashneet-bhatia) [reply yash-datalake/nested/blobTriggerdFunction.json]: This API version was creating problem while running spark job in data factory. Tested all possible API version came out to the conclusion that Storage account and azure functions need to have similar API versions.
That's I roll backed it to the version where both are same

File: yash-datalake/azuredeploy.json (hunk 1 above)
Comment (jashneet-bhatia) [reply yash-datalake/azuredeploy.json]: Newer API versions are creating problem right now

File: yash-datalake/nested/storageAccount.json (hunk 4 above)
Comment (jashneet-bhatia) [reply yash-datalake/nested/storageAccount.json]: This API version was creating problem while running spark job in data factory. Tested all possible API version came out to the conclusion that Storage account and azure functions need to have similar API versions.
That's I roll backed it to the version where both are same

i.e 2018-11-01

**Comment by jashneet-bhatia** [comment]
Hii @bmoore-msft 
**This is utmost important..!**
Can you please update blob having all our artifacts with proper directory structure because steps which will be performed later depends on that a lot 

This is the blob where are current artifacts are uploaded
https://azbotstorage.blob.core.windows.net/sample-artifacts/yash-datalake

In here we want to have a directory structure so that our solution works  properly!

Base URL: https://azbotstorage.blob.core.windows.net/sample-artifacts/yash-datalake

Relative URLs:

1. datasets/customers/customerData.csv
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/customers/customerData.csv

2. datasets/demographics/demographics_data_20170520.csv
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/demographics/demographics_data_20170520.csv
3. datasets/orders/order_data.csv
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/orders/order_data.csv

4. datasets/products/products.csv2013
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/products/products.csv2013

5. datasets/products/products.csv2014
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/products/products.csv2014

6. datasets/products/products.csv2015
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/products/products.csv2015

7. datasets/products/products.csv2016
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/datasets/products/products.csv2016

8. jars/Neo4JSparkTransformation-0.0.1-SNAPSHOT-jar-with-dependencies.jar
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/jars/Neo4JSparkTransformation-0.0.1-SNAPSHOT-jar-with-dependencies.jar (10.42 MiB)

9. scripts/Neo4JSparkTransformation-0.0.1-SNAPSHOT-jar-with-dependencies.jar
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/scripts/Neo4JSparkTransformation-0.0.1-SNAPSHOT-jar-with-dependencies.jar (112.33 MiB)

10. customer_updated_refrence.json
Download URL- https://yashdatalake.blob.core.windows.net/yashdatalakeresource/customer_updated_refrence.json


**Please upload all of these resources carefully at with there respective directory structure**

File: yash-datalake/azuredeploy.json (hunk 5 above)
Comment (jashneet-bhatia) [reply yash-datalake/azuredeploy.json]: Yes done with required changes

**Comment by bmoore-msft** [comment]
@jashneet-bhatia - the new artifacts are here:
https://azbotstorage.blob.core.windows.net/sample-artifacts/yash-datalake-2

If needed we can remove the old set once we merge these changes.

**Comment by jashneet-bhatia** [comment]
@bmoore-msft Yes you can delete those older artifacts after merging these changes

I have updated blob link in main template and its working fine. So now it is ready for the merge.

Thank you!

File: yash-datalake/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply yash-datalake/azuredeploy.json]: What's the problem?  If there's a bug in the template or the platform we should address it...

Comment (jashneet-bhatia) [reply yash-datalake/azuredeploy.json]: Actually this version was suggested in best practices check in devops pipeline so i used it but later when I checked in docs this version wasn't
So I just reverted to the next Lower version


https://docs.microsoft.com/en-us/azure/templates/microsoft.resources/2018-05-01/deployments

**Pull request merged**